### ISSUE TITLE
Build errors against v14

### ISSUE THREAD

**Issue by devrimgunduz-edb** [tada/pljava#359]
Hi,

1.6.2 throws error while building against v14. Can you please take a look? Release is due next Thursday.

Regards, Devrim

> [INFO] Using compiling/linking rules for Linux
In file included from /usr/pgsql-14/include/server/postgres.h:46,
                 from /build/pljava-1_6_2/pljava-so/src/main/c/TypeOid.c:13:
/build/pljava-1_6_2/pljava-so/src/main/c/TypeOid.c: In function 'Java_org_postgresql_pljava_jdbc_TypeOid__1dummy':
/build/pljava-1_6_2/pljava-so/src/main/c/TypeOid.c:55:22: error: 'PGNODETREEOID' undeclared (first use in this function); did you mean 'PG_NODE_TREEOID'?
   55 |         CONFIRMCONST(PGNODETREEOID);
      |                      ^~~~~~~~~~~~~
/usr/pgsql-14/include/server/c.h:912:29: note: in definition of macro 'StaticAssertStmt'
  912 |         do { _Static_assert(condition, errmessage); } while(0)
      |                             ^~~~~~~~~
/build/pljava-1_6_2/pljava-so/src/main/c/TypeOid.c:55:9: note: in expansion of macro 'CONFIRMCONST'
   55 |         CONFIRMCONST(PGNODETREEOID);
      |         ^~~~~~~~~~~~
/build/pljava-1_6_2/pljava-so/src/main/c/TypeOid.c:55:22: note: each undeclared identifier is reported only once for each function it appears in
   55 |         CONFIRMCONST(PGNODETREEOID);
      |                      ^~~~~~~~~~~~~
/usr/pgsql-14/include/server/c.h:912:29: note: in definition of macro 'StaticAssertStmt'
  912 |         do { _Static_assert(condition, errmessage); } while(0)
      |                             ^~~~~~~~~
/build/pljava-1_6_2/pljava-so/src/main/c/TypeOid.c:55:9: note: in expansion of macro 'CONFIRMCONST'
   55 |         CONFIRMCONST(PGNODETREEOID);
      |         ^~~~~~~~~~~~
/build/pljava-1_6_2/pljava-so/src/main/c/TypeOid.c:26:18: error: expression in static assertion is not an integer
   26 | StaticAssertStmt((c) == (org_postgresql_pljava_jdbc_TypeOid_##c), \
      |                  ^
/usr/pgsql-14/include/server/c.h:912:29: note: in definition of macro 'StaticAssertStmt'
  912 |         do { _Static_assert(condition, errmessage); } while(0)
      |                             ^~~~~~~~~
/build/pljava-1_6_2/pljava-so/src/main/c/TypeOid.c:55:9: note: in expansion of macro 'CONFIRMCONST'
   55 |         CONFIRMCONST(PGNODETREEOID);
      |         ^~~~~~~~~~~~
/build/pljava-1_6_2/pljava-so/src/main/c/VarlenaWrapper.c: In function 'pljava_VarlenaWrapper_Input':
/build/pljava-1_6_2/pljava-so/src/main/c/VarlenaWrapper.c:222:48: error: 'struct varatt_external' has no member named 'va_extsize'; did you mean 'va_rawsize'?
  222 |                         parked = toast_pointer.va_extsize + VARHDRSZ;
      |                                                ^~~~~~~~~~
      |                                                va_rawsize
/build/pljava-1_6_2/pljava-so/src/main/c/type/ErrorData.c: In function 'Java_org_postgresql_pljava_internal_ErrorData__1isShowFuncname':
/build/pljava-1_6_2/pljava-so/src/main/c/type/ErrorData.c:246:52: error: 'ErrorData' has no member named 'show_funcname'; did you mean 'funcname'?
  246 |         return (jboolean)((ErrorData*)p2l.ptrVal)->show_funcname;
      |                                                    ^~~~~~~~~~~~~
      |                                                    funcname
/build/pljava-1_6_2/pljava-so/src/main/c/type/ErrorData.c:247:1: warning: control reaches end of non-void function [-Wreturn-type]
  247 | }
      | ^
/build/pljava-1_6_2/pljava-so/src/main/c/type/SQLXMLImpl.c: In function '_SQLXML_canReplaceType':
/build/pljava-1_6_2/pljava-so/src/main/c/type/SQLXMLImpl.c:54:39: error: 'PGNODETREEOID' undeclared (first use in this function); did you mean 'PG_NODE_TREEOID'?
   54 |                 Type_getOid(other) == PGNODETREEOID  ||  /* a synthetic rendering */
      |                                       ^~~~~~~~~~~~~
      |                                       PG_NODE_TREEOID
/build/pljava-1_6_2/pljava-so/src/main/c/type/SQLXMLImpl.c:54:39: note: each undeclared identifier is reported only once for each function it appears in
/build/pljava-1_6_2/pljava-so/src/main/c/type/SQLXMLImpl.c: In function '_SQLXML_obtain':
/build/pljava-1_6_2/pljava-so/src/main/c/type/SQLXMLImpl.c:143:14: error: 'PGNODETREEOID' undeclared (first use in this function); did you mean 'PG_NODE_TREEOID'?
  143 |         case PGNODETREEOID:
      |              ^~~~~~~~~~~~~
      |              PG_NODE_TREEOID
/build/pljava-1_6_2/pljava-so/src/main/c/type/SQLXMLImpl.c: In function '_SQLXML_canReplaceType':
/build/pljava-1_6_2/pljava-so/src/main/c/type/SQLXMLImpl.c:57:1: warning: control reaches end of non-void function [-Wreturn-type]
   57 | }
      | ^
/build/pljava-1_6_2/pljava-so/src/main/c/type/Type.c: In function 'TypeClass_alloc2':
/build/pljava-1_6_2/pljava-so/src/main/c/type/Type.c:1000:33: warning: cast between incompatible function types from 'void (*)(struct PgObject_ *)' to 'jvalue (*)(struct Type_ *, Datum)' {aka 'jvalue (*)(struct Type_ *, long unsigned int)'} [-Wcast-function-type]
 1000 |         self->coerceDatum     = (DatumCoercer)_PgObject_pureVirtualCalled;
      |                                 ^
/build/pljava-1_6_2/pljava-so/src/main/c/type/Type.c:1001:33: warning: cast between incompatible function types from 'void (*)(struct PgObject_ *)' to 'Datum (*)(struct Type_ *, struct _jobject *)' {aka 'long unsigned int (*)(struct Type_ *, struct _jobject *)'} [-Wcast-function-type]
 1001 |         self->coerceObject    = (ObjectCoercer)_PgObject_pureVirtualCalled;
      |                                 ^
/build/pljava-1_6_2/pljava-so/src/main/c/type/Type.c: In function 'Type_registerType':
/build/pljava-1_6_2/pljava-so/src/main/c/type/Type.c:1087:57: warning: cast between incompatible function types from 'void (*)(struct PgObject_ *)' to 'struct Type_ * (*)(Oid)' {aka 'struct Type_ * (*)(unsigned int)'} [-Wcast-function-type]
 1087 |         _registerType(type->typeId, javaTypeName, type, (TypeObtainer)_PgObject_pureVirtualCalled);
      |                                                         ^
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for PostgreSQL PL/Java 1.6.2:
[INFO] 
[INFO] PostgreSQL PL/Java ................................. SUCCESS [  0.824 s]
[INFO] PL/Java API ........................................ SUCCESS [  2.867 s]
[INFO] PL/Java backend Java code .......................... SUCCESS [  2.394 s]
[INFO] PL/Java PGXS ....................................... SUCCESS [  2.036 s]
[INFO] PL/Java backend native code ........................ FAILURE [  7.314 s]
[INFO] PL/Java Ant tasks .................................. SKIPPED
[INFO] PL/Java examples ................................... SKIPPED
[INFO] PL/Java packaging .................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  15.523 s
[INFO] Finished at: 2021-09-21T18:36:05+01:00
[INFO] ---------------------------------------------------------------

**Comment by jcflack** [comment]
Believed resolved in 1.6.3.
